backend/routes/meta.js

```diff
@@ -119,18 +118,107 @@
 	router.get("/revision/master/head/meta/all.json", middlewares.hasReadAccessToModel, getAllMetadata);
 
 	/**
-	 * @api {get} /:teamspace/:model/revision/:rev/meta/all.json Get all meta data by revision
+	 * @api {get} /:teamspace/:model/revision/:rev/meta/all.json Get all metadata by revision
 	 * @apiName getAllMetadataByRev
 	 * @apiGroup Meta
 	 * @apiDescription Get all tree objects with their metadata tags by revision.
 	 * See more details <a href='#api-Meta-getAllMetadata'>here</a>.
 	 *
 	 * @apiParam {String} teamspace Name of teamspace
 	 * @apiParam {String} model Model ID
-	 * @apiParam {String} rev Revision to get meta data from
+	 * @apiParam {String} rev Revision to get metadata from
 	 */
 	router.get("/revision/:rev/meta/all.json", middlewares.hasReadAccessToModel, getAllMetadata);
 
+	/**
+	 * @api {post} /:teamspace/:model/revision(/master/head/|/:revId)/meta/rules Filter metadata by rules
+	 * @apiName queryMetadataByRules
+	 * @apiGroup Meta
+	 * @apiDescription Get all objects matching filter rules in the tree with their metadata.
+	 *
+	 * @apiParam {String} teamspace Name of teamspace
+	 * @apiParam {String} model Model ID
+	 * @apiParam (Query) {Boolean} [meshids] Flag that returns Mesh IDs for matching rule queries
+	 *
+	 * @apiExample {post} Example usage (/master/head)
+	 * POST /teamSpace1/3549ddf6-885d-4977-87f1-eeac43a0e818/revision/master/head/meta/rules HTTP/1.1
+	 *
+	 * @apiExample {post} Example usage (/:revId)
+	 * POST /teamSpace1/3549ddf6-885d-4977-87f1-eeac43a0e818/revision/00000000-0000-0000-0000-000000000001/meta/rules HTTP/1.1
+	 *
+	 * @apiExample {post} Example usage (mesh IDs)
+	 * POST /teamSpace1/3549ddf6-885d-4977-87f1-eeac43a0e818/revision/master/head/meta/rules?meshids=true HTTP/1.1
+	 *
+	 * @apiSuccessExample {json} Success (metadata):
+	 * {
+	 *    "data": [
```

Comment:
I think we may need to change this return to tell the user which model it came from... If this endpoint points to a federation, the metadata are always in the submodels, and at the moment there's no way to tell where it came from with this schema.

thoughts @mosment3DRepo ?

Reply:
@carmenfan @mosment3DRepo At the moment, it is supposed to match the response for getAllMetadata. For federations, it should return information about the submodel. Maybe I should make this clearer in the documentation.

A federation response will look a bit like this:
```
{
    "data": [],
    "subModels": [
        {
            "data": [
                {
                    "_id": "01bbf2d0-a9a1-4593-b8a7-e185a791ffdb",
                    "parents": [
                        "73fe5634-fb56-481d-95f8-3666bcc65c17"
                    ],
                    "metadata": {
                        "IFC Type": "IfcBuildingElementProxy",
                        "IFC GUID": "1HOEJRLv561Qi4FACVNHPH",
                        "Reference": "Family2",
                        "Host": "Level : Level 1",
                        "Level": "Level: Level 1",
                        "Moves With Nearby Elements": "False",
                        "Offset": 1432.044264,
                        "Area": 2767.039224,
                        "Volume": 4504.122208,
                        "Category": "Site",
                        "Family": "Family2: Family2",
                        "Family and Type": "Family2: Family2",
                        "Type": "Family2: Family2",
                        "Type Id": "Family2: Family2",
                        "Phase Created": 1,
                        "Type Name": "Family2",
                        "Family Name": "Family2"
                    }
                }
            ],
            "account": "charence",
            "model": "6275729d-d772-4e21-a60f-eb6eb59360de"
        }
    ]
}
```